modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/**
+ * Checks if element is disabled or whether it exists in a disabled element tree
+ * @param {!Element} element The DOM element to check.
+ * @return {boolean}
+ */
+function isDisabled(node) {
+	return !!(node.disabled || node.closest('[disabled]'));
+}
+
+/**
+ * Listens to the specified event on the given DOM element, but only calls the
+ * given callback listener when it's triggered by elements that match the
+ * given selector or target element.
+ * @param {!Element} element The DOM element the event should be listened on.
+ * @param {string} eventName The name of the event to listen to.
+ * @param {!Element|string} selectorOrTarget Either an element or css selector
+ *     that should match the event for the listener to be triggered.
+ * @param {!function(!Object)} callback Function to be called when the event
+ *     is triggered. It will receive the normalized event object.
+ * @return {!EventHandle} Can be used to remove the listener.
+ */
+function delegate(element, eventName, selectorOrTarget, callback) {
+	const eventHandler = (event) => {
+		if (eventName === 'click' && isDisabled(event.target)) {
+			return;
+		}
+
+		if (
+			event.target === selectorOrTarget ||
```

Comment:
I know I was the one that copied over the definition for `selectorOrTarget` from https://github.com/metal/metal.js/blob/master/packages/metal-dom/src/domNamed.js#L233 but I wonder if this makes sense at all.

On a quick inspection I didn't see us passing any specific targets and I don't really understand why we'd want to in the context of a delegate event.

I'd say we could simplify and always assume a string selector.

Reply:
Oh good point, I didn't really think about that. I will move it to just "selector". My guess is that metal-dom had that for something specific in metal.js.